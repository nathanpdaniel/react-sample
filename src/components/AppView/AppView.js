import React from 'react';
import AppForm from '../AppForm/AppForm';
import { AppValues } from '../AppValues/AppValues';

export class AppView extends React.Component {
  state = {
    valuesSubmitted: false,
    values: null,
  }

  handleSubmit = (values) => {
    // convert values to numeric.  Typically, we'd be sending to an API
    const convertedValues = {
      sampleSize: Number(values.sampleSize),
      sampleMean: Number(values.sampleMean),
      standardDeviation: Number(values.standardDeviation),
      testHypothesis: values.testHypothesis || false,
      hypothesizedMean: values.hypothesizedMean ? Number(values.hypothesizedMean) : null
    }

    this.setState({ valuesSubmitted: true, values: convertedValues })  
  }

  handleBack = () => {
    this.setState({ valuesSubmitted: false, values: null });
  }

  render() {
    const { valuesSubmitted, values } = this.state;

    if (valuesSubmitted) {
      return <AppValues values={values} onBack={this.handleBack} />
    }

    return <AppForm onSubmit={this.handleSubmit} {...this.props}></AppForm>
  }
}