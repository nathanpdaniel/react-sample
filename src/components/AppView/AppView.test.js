import React from 'react';
import { shallow } from 'enzyme';
import { AppView } from './AppView';

describe('AppValues Tests', () => {
  it('should display properly before form submission', () => {
    const component = shallow(<AppView />);
    expect(component).toMatchSnapshot();
  });
  it('should display properly after form submission', () => {
    const state = {
      valuesSubmitted: true,
      values: {
        sampleSize: 12,
        sampleMean: 1.2,
        standardDeviation: 2.3,
        testHypothesis: false,
      }
    }
    const component = shallow(<AppView />);
    component.setState(state, () => {
      expect(component).toMatchSnapshot();
    });
  });
});