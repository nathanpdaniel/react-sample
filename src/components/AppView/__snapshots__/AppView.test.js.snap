// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppValues Tests should display properly after form submission 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onBack": [Function],
      "values": Object {
        "sampleMean": 1.2,
        "sampleSize": 12,
        "standardDeviation": 2.3,
        "testHypothesis": false,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onBack": [Function],
        "values": Object {
          "sampleMean": 1.2,
          "sampleSize": 12,
          "standardDeviation": 2.3,
          "testHypothesis": false,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AppValues Tests should display properly before form submission 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "appForm",
      "getFormState": [Function],
      "keepDirtyOnReinitialize": false,
      "onSubmit": [Function],
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "appForm",
        "getFormState": [Function],
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "displayName": "ReduxForm",
      "render": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "appForm",
        "getFormState": [Function],
        "keepDirtyOnReinitialize": false,
        "onSubmit": [Function],
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "destroyOnUnmount": true,
          "enableReinitialize": false,
          "forceUnregisterOnUnmount": false,
          "form": "appForm",
          "getFormState": [Function],
          "keepDirtyOnReinitialize": false,
          "persistentSubmitErrors": false,
          "pure": true,
          "shouldAsyncValidate": [Function],
          "shouldError": [Function],
          "shouldValidate": [Function],
          "shouldWarn": [Function],
          "touchOnBlur": true,
          "touchOnChange": false,
          "updateUnregisteredFields": false,
        },
        "displayName": "ReduxForm",
        "render": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
